.github/workflows/gradle-build-pr.yml

```diff
@@ -96,16 +103,50 @@ jobs:
           # export the original IP list so it can be restored during cleanup
           RESTORE_IPS=$(printf '%s\n' $CUR_IPS | jq -s -r 'add | unique | join(",")')
           echo "::set-env name=RESTORE_IPS::${RESTORE_IPS}"
-      # build and push container to integration
-      - name: "Set up integration environment"
+      - name: Check for available namespace to run tests in
+        run: |
+          NAMESPACES=("integration-1" "integration-2" "integration-3" "integration-4" "integration-5")
+          find_namespace ()
+          {
+            for i in "${NAMESPACES[@]}"
+            do
+              if kubectl get secrets -n ${i} ${i}-inuse > /dev/null 2>&1; then
+                printf "Namespace ${i} in use Skipping\n"
+              else
+                printf "Namespace ${i} not in use Deploying integration test to ${i}\n"
+                kubectl create secret generic ${i}-inuse --from-literal=inuse=${i} -n ${i}
+                tail=$(echo $i | awk -F- {'print $2'})
+                echo "::set-env name=IT_JADE_API_URL::https://jade-${tail}.datarepo-integration.broadinstitute.org"
+                echo "::set-env name=NAMESPACEINUSE::${i}"
+                return 0
+              fi
+            done
+            sleep 60
+            find_namespace
```

Comment:
Perhaps we might want to retry only a few times instead of infinitely?

Reply:
It seems fine on infinite worst case the job is killed in 180 minutes.


This deploy was in queue for 30 minutes ish. 
https://github.com/DataBiosphere/jade-data-repo/runs/499537076?check_suite_focus=true

Reply:
Given that there's no way to retry or handle unaddressed events yet, I think we'll want to poll until Github kills the action after 3 hours.